parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -160,7 +160,9 @@ API_SERVER_DNS_RETRIES=20
 if [[ $API_SERVER_NAME == *.privatelink.* ]]; then
   API_SERVER_DNS_RETRIES=200
 fi
+{{- if not EnableHostsConfigAgent}}
 RES=$(retrycmd_if_failure ${API_SERVER_DNS_RETRIES} 1 3 nslookup ${API_SERVER_NAME})
+{{- end}}
 STS=$?
```

Comment:
This makes $STS depend on a previous operation, it may be fine. But to be safer can we do
{{- if not EnableHostsConfigAgent}}
RES=$(retrycmd_if_failure ${API_SERVER_DNS_RETRIES} 1 3 nslookup ${API_SERVER_NAME})
STS=$?
{{- else}}
STS=0
{{- end}}

Reply:
makes sense. thanks